frontend/src/modules/readsets/reducers.ts

```diff
@@ -0,0 +1,51 @@
+
+import { Readset } from "../../models/frontend_models";
+import { indexByID } from "../../utils/objects";
+import READSETS from "./actions"
+import { merge } from "object-path-immutable"
+
+export const readsets = (
+    state = {
+        itemsByID: {},
+        items: [],
+        isFetching: false,
+    }
+    ,
+    action) => {
+    switch (action.type) {
+        case READSETS.GET.REQUEST:
+            return merge(state, ['itemsByID', action.meta.id], { id: action.meta.id, isFetching: true });
+        case READSETS.GET.RECEIVE:
+            return merge(state, ['itemsByID', action.meta.id], { ...action.data, isFetching: false });
```

Comment:
you will likely override withMetrics readsets without readset if you do not have the same mechanism to deal metrics in your get as in your list.